### PR TITLE
Update route-agent Pod env with the host nodeName

### PR DESCRIPTION
**Pull request by sridhargaddam**
Related to: https://github.com/submariner-io/submariner-operator/issues/835
Signed-Off-by: Sridhar Gaddam <sgaddam@redhat.com>

### CODE REVIEW

**Review by tpantelis — DISMISSED** [review]

File: submariner/templates/route-agent-ds.yaml [hunk 1]
```diff
@@ -41,10 +41,14 @@ spec:
           value: "{{ .Values.submariner.clusterId }}"
         - name: SUBMARINER_DEBUG
           value: "{{ .Values.submariner.debug }}"
-        - name: SUBMARINER_CLUSTERCIDR                                                                                                                                                        
+        - name: SUBMARINER_CLUSTERCIDR
           value: "{{ .Values.submariner.clusterCidr }}"
         - name: SUBMARINER_SERVICECIDR
           value: "{{ .Values.submariner.serviceCidr }}"
+        - name: NODE_NAME
```
Comment (tpantelis) [inline submariner/templates/route-agent-ds.yaml:48]: Perhaps we should prefix with _SUBMARINER__ to be consistent?

Comment (sridhargaddam) [reply submariner/templates/route-agent-ds.yaml:48]: Well, this is not a Submariner parameter and it just points to the actual NODE_NAME. So, I'm not sure if we want to use SUBMARINER_ here.

Comment (sridhargaddam) [reply submariner/templates/route-agent-ds.yaml:48]: IMHO SUBMARINER_NODE_NAME will actually be confusing to the users as it points to the actual node and every route agent has a different value (i.e., it points to the nodeName on which its running)

Comment (tpantelis) [reply submariner/templates/route-agent-ds.yaml:48]: So this won't be consumed by our code?

Comment (tpantelis) [reply submariner/templates/route-agent-ds.yaml:48]: As discussed on chat, this is fine.

**Review by tpantelis — APPROVED** [review]

**Review by mangelajo — APPROVED** [review]

**Pull request merged**